packages/schematics/src/command-line/affected.ts

```diff
@@ -126,21 +145,32 @@ function printError(e: any) {
 }
 
 function build(
-  apps: string[],
+  projects: string[],
   parsedArgs: YargsAffectedOptions,
   workspaceResults: WorkspaceResults
 ) {
-  if (apps.length > 0) {
+
+  const depGraph = readDepGraph();
+  const sortedProjects = topologicallySortProjects(depGraph);
+  const sortedAffectedProjects = sortedProjects.filter(
+    pp => projects.indexOf(pp) > -1
+  );
+  const projectsToBuild = sortedAffectedProjects.filter(p => {
```

Comment:
This function is meant (currently anyways) to build all apps it is passed with no opinions.

This logic is better placed within `shared.ts` where the rest of the logic lives. See other comment.